protocols/raft/src/main/java/io/atomix/protocols/raft/roles/FollowerRole.java

```diff
@@ -103,9 +103,14 @@ private void resetHeartbeatTimeout() {
     Duration delay = raft.getElectionTimeout().plus(Duration.ofMillis(random.nextInt((int) raft.getElectionTimeout().toMillis())));
     heartbeatTimer = raft.getThreadContext().schedule(delay, () -> {
       heartbeatTimer = null;
-      if (isRunning() && (raft.getFirstCommitIndex() == 0 || raft.getState() == RaftContext.State.READY)) {
+      if (isRunning() && (raft.getFirstCommitIndex() == 0
+          || raft.getState() == RaftContext.State.READY)) {
+        final long missTime = System.currentTimeMillis() - lastHeartbeat;
+        log.info("No heartbeat from {} in the last {} (calculated from last {} ms)",
+            raft.getLeader(), delay, missTime);
+        raft.getRaftRoleMetrics().countHeartbeatMiss(lastHeartbeat);
+
```

Comment:
Should we reset `lastHeartBeat` here? Or are we sure we will transition to a new role so `FollowerRole` will be reinstantiated?

Reply:
Then we will see a very large heartbeat interval, no? Wouldn't that create some false positives for the heatmap? Maybe since it's only one it doesn't make much of a difference :thinking: